### PR TITLE
Possiblity to share remote state dict

### PR DESCRIPTION
**Pull request by gmuraru**
## Description
Share a model using a remote dictionary for `weight` and `bias`
Fixes #91 

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Added tests

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Review by aanurraj — APPROVED** [review]
👍🏻

File: src/sympc/module/nn/conv.py [hunk 1]
```diff
@@ -74,20 +75,36 @@ def forward(self, x: MPCTensor) -> MPCTensor:
...
+    ) -> None:
         """Share the parameters of the normal Conv2d layer.
 
         Args:
             state_dict (Dict[str, Any]): the state dict that would be shared
         """
+
+        bias = None
+        if ispointer(state_dict):
+            weight = state_dict["weight"].resolve_pointer_type()
```
Comment (LaRiffle) [inline src/sympc/module/nn/conv.py:90]: sorry for the stupid Q: what does `resolve_pointer_type` do?

File: src/sympc/module/__init__.py [hunk 2]
```diff
@@ -7,23 +7,26 @@
...
 from .nn import Conv2d
 from .nn import Linear
 
-MAP_TORCH_TO_SYMPC = {torch.nn.Linear: Linear, torch.nn.Conv2d: Conv2d}
+MAP_TORCH_TO_SYMPC = {
+    "Linear": Linear,
+    "Conv2d": Conv2d,
+    "Conv2dPointer": Conv2d,
+    "LinearPointer": Linear,
+}
```
Comment (LaRiffle) [inline src/sympc/module/__init__.py:26]: nit: maybe add dynamically the Pointer versions?

**Review by LaRiffle — APPROVED** [review]
I have some (stupid) questions, but apart from this, it's really good! 🙌

File: src/sympc/module/nn/conv.py [hunk 3]
```diff
@@ -74,20 +75,36 @@ def forward(self, x: MPCTensor) -> MPCTensor:
...
+            if "bias" in state_dict.keys().get():
+                bias = state_dict["bias"].resolve_pointer_type()
+            shape = weight.client.python.Tuple(weight.shape)
+            shape = shape.get()
```
[the remark this answers is not in the document]
Comment (gmuraru) [reply src/sympc/module/nn/conv.py:94]: The ```weight.shape``` is a ```torch.Size``` and there is no possibility to serialize the size at this moment :(

File: src/sympc/module/nn/conv.py (hunk 1 above)
Comment (gmuraru) [reply src/sympc/module/nn/conv.py:90]: state_dict is an ```OrderedDictPointer``` and doing a ```get``` on it we do not know what we get back (we get back a generic type named ```AnyPointer```). To be able to call ```shape``` we need to solve its type (```resolve_pointer_type``` would give back a ```TensorPointer```)

File: src/sympc/module/__init__.py (hunk 2 above)
Comment (gmuraru) [reply src/sympc/module/__init__.py:26]: Good idea

**Comment by gmuraru** [comment]
> I have some (stupid) questions, but apart from this, it's really good!

They are not stupid :D

File: src/sympc/module/nn/conv.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (LaRiffle) [reply src/sympc/module/nn/conv.py:94]: Aaaaah ok!

File: src/sympc/module/nn/conv.py (hunk 1 above)
Comment (LaRiffle) [reply src/sympc/module/nn/conv.py:90]: Hmmm I see :)
